proto/buf/alpha/registry/v1alpha2/module.proto

```diff
@@ -0,0 +1,291 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package buf.alpha.registry.v1alpha2;
+
+import "buf/alpha/registry/v1alpha1/user.proto";
+import "buf/alpha/registry/v1alpha2/role.proto";
+import "google/protobuf/timestamp.proto";
+
+enum Visibility {
+  VISIBILITY_UNSPECIFIED = 0;
+  VISIBILITY_PUBLIC = 1;
+  VISIBILITY_PRIVATE = 2;
+}
+
+message Module {
```

Comment:
We already have something named `Module`, this should be something different, but I don't know what

Reply:
Isn't this what we want to change? Repository -> Module, and Module -> Snapshot?

Because we already have both a `module/v1alpha1/module.proto` and a `repository/v1alpha1/module.proto` (and the "repository" version has `Local` varieties of the "module" version, which I personally find confusing at a first glance :).

We could go with `RegistryModuleService` or something of the sorts, I guess.

Reply:
though `RegistryModuleService` would probably be a bad naming practice/redundant since we would arledy be referencing this with something like `registry.v1alpha2.RegistryModuleService`

Reply:
I went with `RemoteModule` (thought about `HostedModule` which sounded less explicit, and `RemoteModuleRepository` which is kinda too verbose). Lmk what you think